chain-abci/src/enclave_bridge/edp/tdbe.rs

```diff
@@ -0,0 +1,238 @@
+            local_listen_address: tdbe_config.local_listen_address.clone(),
+        })
+    }
+
+    pub fn spawn(self) {
+        thread::spawn(move || {
+            let mut device = Device::new()
+                .expect("SGX device was not found")
+                .einittoken_provider(AesmClient::new())
+                .build();
+            let mut enclave_builder = EnclaveBuilder::new("tdb-enclave-app.sgxs".as_ref());
+
+            enclave_builder
+                .coresident_signature()
+                .expect("Enclave signature file not found");
+            enclave_builder.usercall_extension(self);
+
+            let enclave = enclave_builder
+                .build(&mut device)
+                .expect("Failed to build enclave");
+            enclave.run().expect("Failed to start enclave")
+        });
+    }
+}
+
+/// Fetches connection details of remote TDBE server using TM RPC
+fn fetch_remote_tdbe_connection_details(
+    _tdbe_config: &TdbeConfig,
+) -> std::io::Result<(Option<String>, Option<String>)> {
+    // TODO: Fetch connection details using TM RPC (use `client-common::TendermintRpcClient`?)
+    Ok((None, None))
+}
+
+fn spawn_chain_abci_thread(mut receiver: UnixStream, storage: Arc<dyn KeyValueDB>) {
+    let _ = thread::spawn(move || {
+        let storage = chain_storage::ReadOnlyStorage::new_db(storage);
+
+        while let Ok(enclave_request) = EnclaveRequest::read_from(&mut receiver) {
+            match enclave_request {
+                EnclaveRequest::GetSealedTxData { txids } => {
+                    let response =
+                        EnclaveResponse::GetSealedTxData(get_sealed_tx_data(txids, &storage));
+                    response
+                        .write_to(&mut receiver)
+                        .expect("Unable to write to TDBE <-> chain-abci unix stream");
+                }
+                _ => unreachable!("TDBE can only send `GetSealedTxData` request"),
+            }
+        }
+    });
+}
+
+fn get_sealed_tx_data(txids: Vec<TxId>, storage: &ReadOnlyStorage) -> Option<Vec<SealedLog>> {
+    let mut result = Vec::with_capacity(txids.len());
+
+    for txid in txids {
+        if let Some(txin) = storage.get_sealed_log(&txid) {
+            result.push(txin);
+        } else {
+            return None;
+        }
+    }
+
+    Some(result)
+}
+
+fn spawn_persistence_thread(mut receiver: UnixStream, storage: Arc<dyn KeyValueDB>) {
+    let _ = thread::spawn(move || {
+        let storage = chain_storage::Storage::new_db(storage);
+        let mut buffer = HashMap::new();
+        let mut kvdb = chain_storage::buffer::BufferStore::new(&storage, &mut buffer);
+
+        while let Ok(persistence_command) = PersistenceCommand::read_from(&mut receiver) {
+            chain_storage::store_sealed_log(
+                &mut kvdb,
+                &persistence_command.transaction_id,
+                &persistence_command.sealed_log,
+            );
+        }
+    });
+}
```

Comment:
1. BufferStore won't persist it until it's flushed
2. how to signal when this is finished? otherwise this will loop forever ?
3. how to signal `last_fetched_block` back to chain-abci? https://github.com/crypto-com/chain-docs/blob/master/docs/modules/tdbe.md#catch-up-process

Reply:
Ok. I'll convert `PersistenceCommand` to an enum and add a `Finish` variant which will be sent when all the transactions are persisted? For flushing, is it ok to flush after each update?

Reply:
I think it's best to flush at the end -- @yihuang may comment

Reply:
Done